src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -191,67 +191,95 @@ export default function CIPipeline({
+            setPresetPlugins(_presetPlugin)
+            setSharedPlugins(_sharedPlugin)
+            await getMandatoryPluginData(_formData, [..._presetPlugin, ..._sharedPlugin])
         }
-    }, [location.pathname, ciPipeline.pipelineType])
+        catch (error) {
+            showError(error)
+        }
+    }
 
-    const getEnvironments = (envId) => {
+    const getEnvironments = async (envId: number): Promise<void> => {
         envId = envId || 0
-        getEnvironmentListMinPublic()
-            .then((response) => {
-                let list = []
-                list.push({
-                    id: 0,
-                    clusterName: '',
-                    clusterId: null,
-                    name: DEFAULT_ENV,
-                    active: false,
-                    isClusterActive: false,
-                    description: 'System default',
-                })
-                response.result?.forEach((env) => {
-                    if (env.cluster_name !== 'default_cluster' && env.isClusterCdActive) {
-                        list.push({
-                            id: env.id,
-                            clusterName: env.cluster_name,
-                            clusterId: env.cluster_id,
-                            name: env.environment_name,
-                            active: false,
-                            isClusterActive: env.isClusterActive,
-                            description: env.description,
-                        })
-                        const _selectedEnv = list.find((env) => env.id == envId)
-                        setSelectedEnv(_selectedEnv)
-                    }
-                })
-                sortObjectArrayAlphabetically(list, 'name')
-                setEnvironments(list)
+        try {
+            const list = []
+            list.push({
+                id: 0,
+                clusterName: '',
+                clusterId: null,
+                name: DEFAULT_ENV,
+                active: false,
+                isClusterActive: false,
+                description: 'System default',
             })
-            .catch((error) => {
-                showError(error)
+            const environmentResponse = await getEnvironmentListMinPublic()
+            const environmentResult = environmentResponse?.result ?? []
+            environmentResult.forEach((env) => {
+                if (env.cluster_name !== 'default_cluster' && env.isClusterCdActive) {
+                    list.push({
+                        id: env.id,
+                        clusterName: env.cluster_name,
+                        clusterId: env.cluster_id,
+                        name: env.environment_name,
+                        active: false,
+                        isClusterActive: env.isClusterActive,
+                        description: env.description,
+                    })
+                    const _selectedEnv = list.find((env) => env.id == envId)
```

Comment:
Either this find should be outside or there should be a check for envId direct. The setter must be outside of foreach

Reply:
Done